lib/fluent/plugin/out_google_cloud.rb

```diff
@@ -462,11 +456,23 @@ def parse_json_or_nil(input)
       # Only here to please rubocop...
       return nil if input.nil?
 
-      begin
-        return JSON.parse(input)
-      rescue JSON::ParserError
-        return nil
-      end
+      input.each_codepoint do |c|
+        if c == 123
+          # left curly bracket (U+007B)
+          begin
+            return JSON.parse(input)
+          rescue JSON::ParserError
+            return nil
+          end
+        else
+          # Break (and return nil) unless the current character is whitespace,
+          # in which case we continue to look for a left curly bracket.
+          # Whitespace as per the JSON spec are: tabulation (U+0009),
+          # line feed (U+000A), carriage return (U+000D), and space (U+0020).
+          break unless c == 9 || c == 10 || c == 13 || c == 32
```

Comment:
I couldn't find anything other than the regular expression /\s/ which I wanted to avoid because of a fear (perhaps irrational?) of this being less performant than the explicit check. The check also nicely corresponds with what's in the JSON spec.